src/hub_predtimechart/app/generate_target_json_files.py

```diff
@@ -34,17 +35,18 @@ def main(hub_dir, target_out_dir):
     logger.info(f'main({hub_dir=}, {target_out_dir=}): entered')
 
     hub_dir = Path(hub_dir)
+    hub_config = HubConfig(hub_dir, Path(ptc_config_file))
     target_out_dir = Path(target_out_dir)
-    target_data_df = get_target_data_df(hub_dir, 'target-hospital-admissions.csv')
+    target_data_df = get_target_data_df(hub_dir, hub_config.target_data_file_name)
```

Comment:
There was discussion on Slack about making the `target_data_file_name` optional so that a hub without target data could still use the tool.  My understanding is that right now, this `get_target_data_df` call will end up raising an error if `target_data_file_name` is `None` though? Is that right?

Handling this case seems like a low priority to me (we don't currently have users asking for this feature).  Should we file an issue to handle this later, or handle it now?  And if we defer handling it to later, perhaps we should just make the `target_data_file_name` field required for now?

Reply:
Thanks.  What I'm seeing in `get_target_data_df()` is that it catches and then re-raises the `FileNotFoundError`.  Will this ultimately mean that if the ptc config file doesn't contain that field, the generation of target data files will error out?  I guess I was expecting to see some kind of early exit from target file creation if `hub_config.target_data_file_name is None`, but maybe I'm not seeing how the pieces of the full workflow fit together.

Reply:
> Will this ultimately mean that if the ptc config file doesn't contain that field, the generation of target data files will error out? 

I was originally thinking that the `FileNotFoundError` is reasonable, but thinking about it from the perspective of running a pipeline, I would actually want it to emit a message saying that there is no target data to generate instead of an error because when I'm running the pipeline, I want errors to show up for hub dashboards that _do_ specify target data. If they don't have target data to generate and an error shows up, it's just noise.

Reply:
So do you two want me to log an error and exit if the file is missing, but not actually raise an error?

Reply:
My vote would be:

- If a file path for target data was specified but doesn't exist, we should throw an error
- If a file path for target data was not specified (we get `None` here), we gracefully exit without an error.  We were unable to produce target data, but we were not asked to produce target data, so there's not actually an error condition here.

Reply:
@elray1 : That sounds clean.

@zkamvar : For Evan's first bullet, should I log an error and return, but not actually raise an error? For the exit, do you want me to use a Python `return`, or exit with a system error code, e.g., `sys.exit(1)`?

Reply:
I agree with Evan's breakdown.

Reply:
> In both cases I simply return, not sys.exit(). How's that?

Almost! If I understand correctly, both of these will run through without causing a non-zero exit code on the CLI? Instead, if you emit a `logger.error()`, then that should be paired with a `sys.exit(1)`.

Reply:
Makes sense. Fixed.